File: backend/internal/user/types/user_setting_type.go
package types

/*
 * 設定
 * 1. リマインド機能　何日前？　何時に？　メール通知有効/無効？
 *
 *
 *
 */

/*
 * リマインド設定レスポンス
 */
type RemindSettingResponse struct {
	RemindDaysBefore int  `json:"remindDaysBefore"`
	RemindHour       int  `json:"remindHour"`
	IsEmailEnabled   bool `json:"isEmailEnabled"`
}

/*
 * リマインド設定更新リクエスト
 */
type UpdateRemindSettingRequest struct {
	RemindDaysBefore int  `json:"remindDaysBefore" binding:"min=0,max=30"`
	RemindHour       int  `json:"remindHour" binding:"min=0,max=23"`
	IsEmailEnabled   bool `json:"isEmailEnabled"`
}

File: frontend/src/types/user/user_setting_type.ts
/*
 * user設定関連の型
 */

// リマインド設定
export type RemindSetting = {
  remindDaysBefore: number;
  remindHour: number;
  isEmailEnabled: boolean;
};

// リマインド設定更新リクエスト
export type UpdateRemindSettingRequest = {
  remindDaysBefore: number;
  remindHour: number;
  isEmailEnabled: boolean;
};